### DISCUSSION TITLE
Help with creating proper ITraktSyncHistoryPost containing Seasons for Sync.AddWatchedHistoryItemsAsync

### DISCUSSION THREAD

**Discussion by mpfc75** [henrikfroehling/Trakt.NET#218] [Help] [answered]
Hello Henrik! I am still plugging away on my app and have run into a problem understanding PostBuilders and HistoryPosts.
I wrote a method (**ChangeSeasonWatchedStatusInTraktAsync**) that I pass a bool (watched t/f), show slug, and season number and I make a call to Trakt (**Sync.AddWatchedHistoryItemsAsync**) to mark all episodes in the show/season to be watched or unwatched. I wrote the code below and tested it with a show that happened to only have one season and it appeared to work fine but I later discovered that the season information doesn’t have any impact and the result is that all episodes in all seasons were marked watched. So when I thought it was working with the season, it was actually doing the entire show so my error was masked.

Would you be able to provide an example or documentation on how I should be doing this?

```csharp
public static async Task ChangeSeasonWatchedStatusInTraktAsync(bool watched, string watchedShow, int watchedSeason)
{
    TraktResponse<ITraktSyncHistoryPostResponse> myHistoryAddResponse;
    TraktResponse<ITraktSyncHistoryRemovePostResponse> myHistoryDelResponse;

    MyTraktSeasonDB mySeasonFromDB = await SelectSeasonFromDBAsync(watchedShow, watchedSeason);  // returns a cached season from my SQLite DB

    TraktSeason[] seasonCollection = new TraktSeason[1]; // we only need one because we are only changing one season
    seasonCollection[0] = new TraktSeason();
    seasonCollection[0].Title = mySeasonFromDB.Title;
    seasonCollection[0].Number = watchedSeason;
    seasonCollection[0].Ids = new TraktSeasonIds { Trakt = mySeasonFromDB.IdsTrakt };

    try
    {
        if (watched)
        {
            ITraktSyncHistoryPostBuilder postObject = TraktPost.NewSyncHistoryPost().WithShow(
                new TraktShow
                {
                    Ids = new TraktShowIds { Slug = watchedShow },
                    Seasons = seasonCollection
                });
            ITraktSyncHistoryPost myTraktSyncHistoryPost = new TraktSyncHistoryPost();
            myTraktSyncHistoryPost = postObject.Build();
            myHistoryAddResponse = await client.Sync.AddWatchedHistoryItemsAsync(myTraktSyncHistoryPost);
        }
        else … {remaining code}
```
Thanks!

**Comment by henrikfroehling** [comment, marked as answer]
To mark only specific seasons as watched, you can use
`TraktPost.NewSyncHistoryPost().AddShowAndSeasons(show).WithSeasons(1, 3, 5)`.

For your example, this would look like this:
```csharp
ITraktSyncHistoryPost myTraktSyncHistoryPost = TraktPost.NewSyncHistoryPost().AddShowAndSeasons(
    new TraktShow { Ids = new TraktShowIds { Slug = watchedShow } })
    .WithSeasons(watchedSeason)
    .Build();

myHistoryAddResponse = await client.Sync.AddWatchedHistoryItemsAsync(myTraktSyncHistoryPost);
```

**Reply by mpfc75** [reply]
That did it.  Thank you! Much simpler than my version, and it works too. :-)
